crates/runtime/tests/integration_models.rs

```diff
@@ -38,7 +38,10 @@ fn init_tracing(default_level: Option<&str>) -> DefaultGuard {
     tracing::subscriber::set_default(subscriber)
 }
 
-fn init_tracing_with_task_history(default_level: Option<&str>, rt: &Runtime) -> DefaultGuard {
+fn init_tracing_with_task_history(
+    default_level: Option<&str>,
+    rt: &Runtime,
+) -> (DefaultGuard, TracerProvider) {
```

Comment:
TracerProvider is exposed in order to be able to flush telemetry when it is required (ensure `task_history` contains all traces)